File: example/src/main/java/jp/bglb/bonboru/flux/example/MyDispatcher.java
package jp.bglb.bonboru.flux.example;

import jp.bglb.bonboru.flux.Dispatcher;
import jp.bglb.bonboru.flux.example.action.ActionTypes;
import jp.bglb.bonboru.flux.middleware.Middleware;
import jp.bglb.bonboru.flux.reducer.Reducer;
import jp.bglb.bonboru.flux.store.Store;

/**
 * Created by tmasuda on 2016/04/24.
 */
public class MyDispatcher<T> extends Dispatcher<T, ActionTypes> {

  public MyDispatcher(Reducer<T, ActionTypes> reducer, Store<T> store) {
    super(reducer, store);
  }

  public MyDispatcher(Reducer<T, ActionTypes> reducer, Store<T> store,
      Middleware<T, ActionTypes>... middlewares) {
    super(reducer, store, middlewares);
  }
}

File: kotlinexample/src/main/kotlin/com/example/kotlinexample/MyDispatcher.kt
package com.example.kotlinexample

import com.example.kotlinexample.action.MyActionType
import jp.bglb.bonboru.flux.kotlin.Dispatcher
import jp.bglb.bonboru.flux.reducer.Reducer
import jp.bglb.bonboru.flux.store.Store

/**
 * Created by tetsuya on 2016/06/25.
 */
class MyDispatcher<T>(reducer: Reducer<T, MyActionType>, store: Store<T>) : Dispatcher<T, MyActionType>(
    reducer, store) {
}
